### PR TITLE
Add missing global 'security' declaration to openapi.proto

### PR DESCRIPTION
**Pull request by holgerstolzenberg**
Fixes ApiKeyAuth method being used by clients generated via OpenAPI generator.

Original idea: https://github.com/authzed/authzed-go/pull/255

### CODE REVIEW

**Comment by tstirrat15** [comment]
Gonna have a look at this today. Thanks for putting it up!

**Comment by tstirrat15** [comment]
How does it differ from what's right above it? https://github.com/authzed/api/pull/120/files#diff-2c89108fef442d02f29499535203f20595b6fdd3cff830a5b3fd40e979eece0bR34-R41

**Comment by holgerstolzenberg** [comment]
Okay maybe I have messed up the syntax a bit. Let's take it different, as I am not familiar with `buf`.

This is what the target swagger.json should look like:

Screenshot 2024-11-08 at 17 33 53

I have updated the PR, hope that makes more sense now.

**Comment by tstirrat15** [comment]
Hmm... the thing I'm getting hung up on is that there's two `security` blocks. Is that correct/expected? Should we change the existing `security` block rather than adding a new one?

**Comment by holgerstolzenberg** [comment]
No I think this is totally fine - even if not being really straight forward at fist. The official docs unfortunately use yaml for examples already.

So you have essentially two elements:

`securityDefinitions` defines what is actually _available_ from your API, but it does not apply that auth method to anywhere.
Here it just says: The API supports an Authentication **named** `ApiKeyAuth` that is of type `apiKey` and will be applied by using a _header_ named `Authorization`.

In the `security` section you just name _where_ your defined security methods are going to be applied by just cross referencing them. Line `3035` means, take the auth method named `ApiKey` and then apply it to all endpoints (`[]`).

Even if that looks unfamiliar, it is confirmed working. So you might want to take a look here:
https://github.com/ewerk/authzed-http-client/blob/fc74dc40f8d81720f976b1d866106d4e8907bfed/.github/workflows/main-build.yml#L34

**Review by vroldanbet — CHANGES_REQUESTED** [review]
Thanks for your contribution, I have two comments, otherwise LGTM

File: authzed/api/v1/openapi.proto
Comment (holgerstolzenberg) [reply authzed/api/v1/openapi.proto:43]: Repository local `.gitignore` dropped, as request, though I do not like the global config approach. Have it in the repo, you have it controlled for everyone contributing, no matter what their env is. Even if that means duplicating the file to each repo.

Description added as requested.

File: .gitignore
Comment (holgerstolzenberg) [reply .gitignore]: Repository local `.gitignore` dropped, as request, though I do not like the global config approach. Have it in the repo, you have it controlled for everyone contributing, no matter what their env is. Even if that means duplicating the file to each repo.

**Review by vroldanbet — APPROVED** [review]

**Comment by holgerstolzenberg** [comment]
@tstirrat15 Could you also please review - I am not able to request that and it seems like a second review is required.

**Comment by tstirrat15** [comment]
Yep, this makes sense now. Thank you!

**Review by tstirrat15 — APPROVED** [review]

**Comment by tstirrat15** [comment]
I checked this out and it looks good. I'm going to go ahead and merge this, and then I'll do the work to pull it into authzed-go.

**Pull request merged**
